core-war/src/main/java/org/silverpeas/web/portlets/ComponentInstancePortlet.java

```diff
@@ -23,30 +23,45 @@
  */
 package org.silverpeas.web.portlets;
 
-import org.silverpeas.core.web.portlets.FormNames;
-import org.silverpeas.core.util.URLUtil;
-import org.silverpeas.core.admin.user.model.UserDetail;
 import org.silverpeas.core.admin.service.OrganizationControllerProvider;
+import org.silverpeas.core.admin.user.model.UserDetail;
 import org.silverpeas.core.util.StringUtil;
+import org.silverpeas.core.util.URLUtil;
+import org.silverpeas.core.web.portlets.FormNames;
 
 import javax.portlet.*;
 import java.io.IOException;
 
 public class ComponentInstancePortlet extends GenericPortlet implements FormNames {
 
+  private static String PREF_INSTANCEID = "instanceId";
+
   @Override
   public void doView(RenderRequest request, RenderResponse response)
       throws PortletException, IOException {
     PortletPreferences pref = request.getPreferences();
-    String instanceId = pref.getValue("instanceId", "");
+    String instanceId = pref.getValue(PREF_INSTANCEID, "");
 
     if (OrganizationControllerProvider.getOrganisationController().isComponentAvailable(instanceId,
         UserDetail.getCurrentRequester().getId())) {
       request.setAttribute("URL", URLUtil.getURL(null, null, instanceId) + "portlet");
+    } else {
+      // get portlet window name to hide it (client side)
+      setPortletWindowName(request);
     }
     include(request, response, "portlet.jsp");
   }
 
+  private void setPortletWindowName(RenderRequest request) {
+    String windowId = request.getWindowID();
+    if (StringUtil.isDefined(windowId)) {
+      String portletWindowName = windowId.substring(windowId.lastIndexOf("|")+1);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Put single-quotes around '|' to use the faster "lastIndexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3027)